router/router.go

```diff
@@ -27,4 +41,72 @@ func SetupRoutes(app *fiber.App) {
 	user.Post("/", handler.CreateUser)
 	user.Patch("/:id", middleware.Protected(), handler.UpdateUser)
 	user.Delete("/:id", middleware.Protected(), handler.DeleteUser)
+
+	app.Get("/", func(c *fiber.Ctx) error {
+		err := middleware.FetchUserData(c, googleOauthConfig)
+
+		if err != nil {
+			fmt.Println("Failed to fetch user data:", err)
+			return c.Status(fiber.StatusOK).JSON(fiber.Map{
+				"status": "failed_to_fetch_user_data",
+			})
+		}
+
+		status, ok := c.Locals("status").(string)
+		if !ok {
+			fmt.Println("Status not found in context or not a string")
+			return c.Status(fiber.StatusOK).JSON(fiber.Map{
+				"status": "status_not_found",
+			})
+		}
+
+		if status == "success" {
+			// Return JSON response for success
+			return c.Status(fiber.StatusOK).JSON(fiber.Map{
+				"status": "user_already_logged_in",
+			})
+		} else {
+			// Return JSON response for failure
+			return c.Status(fiber.StatusForbidden).JSON(fiber.Map{
+				"status": "need_to_log_in",
+			})
+		}
+	})
+
+	app.Get("/auth/google", func(c *fiber.Ctx) error {
```

Comment:
can you use `auth.Get("/google", ...)` instead of `app.Get("/auth/google", ...)`?

Reply:
I believe it's required, thats why I changed existing request structure.